controller_manager/src/controller_manager.cpp

```diff
@@ -53,61 +58,787 @@ ControllerManager::load_controller(
+    controller_interface::ControllerInterfaceSharedPtr ct = get_controller_by_name(controller);
+    if (!ct.get()) {
+      if (strictness == controller_manager_msgs::srv::SwitchController::Request::STRICT) {
+        RCLCPP_ERROR(
+          get_logger(),
+          "Could not stop controller with name '%s' because no controller with this name exists",
+          controller.c_str());
+        stop_request_.clear();
+        return controller_interface::return_type::ERROR;
+      } else {
+        RCLCPP_DEBUG(
+          get_logger(),
+          "Could not stop controller with name '%s' because no controller with this name exists",
+          controller.c_str());
+      }
+    } else {
+      RCLCPP_DEBUG(
+        get_logger(),
+        "Found controller '%s' that needs to be stopped in list of controllers",
+        controller.c_str());
+      stop_request_.push_back(ct);
+    }
+  }
+  RCLCPP_DEBUG(get_logger(), "Stop request vector has size %i", (int)stop_request_.size());
+
+  // list all controllers to start
+  for (const auto & controller : start_controllers) {
+    controller_interface::ControllerInterfaceSharedPtr ct = get_controller_by_name(controller);
+    if (!ct.get()) {
+      if (strictness == controller_manager_msgs::srv::SwitchController::Request::STRICT) {
+        RCLCPP_ERROR(
+          get_logger(),
+          "Could not start controller with name '%s' because no controller with this name exists",
+          controller.c_str());
+        stop_request_.clear();
+        start_request_.clear();
+        return controller_interface::return_type::ERROR;
+      } else {
+        RCLCPP_DEBUG(
+          get_logger(),
+          "Could not start controller with name '%s' because no controller with this name exists",
+          controller.c_str());
+      }
+    } else {
+      RCLCPP_DEBUG(
+        get_logger(), "Found controller '%s' that needs to be started in list of controllers",
+        controller.c_str());
+      start_request_.push_back(ct);
+    }
+  }
+  RCLCPP_DEBUG(get_logger(), "Start request vector has size %i", (int)start_request_.size());
+
+#ifdef TODO_IMPLEMENT_RESOURCE_CHECKING
+  // Do the resource management checking
+  std::list<hardware_interface::ControllerInfo> info_list;
+  switch_start_list_.clear();
+  switch_stop_list_.clear();
+#endif
+
+  // lock controllers
+  std::lock_guard<std::recursive_mutex> guard(rt_controllers_wrapper_.controllers_lock_);
+
+  const std::vector<ControllerSpec> & controllers =
+    rt_controllers_wrapper_.get_updated_list(guard);
+  for (const auto & controller : controllers) {
+    bool in_stop_list = false;
+    for (const auto & request : stop_request_) {
+      if (request == controller.c) {
+        in_stop_list = true;
+        break;
+      }
+    }
+
+    bool in_start_list = false;
+    for (const auto & request : start_request_) {
+      if (request == controller.c) {
+        in_start_list = true;
+        break;
+      }
+    }
```

Comment:
again, I think this should be solved more easily with two consecutive calls to `std::find_if`.

Reply:
Done